packages/react/src/useActionForm.ts

```diff
@@ -101,6 +128,198 @@ const disambiguateDefaultValues = (data: any, initialData: any, action: any) =>
   return result;
 };
 
+const reshapeDataForGraphqlApi = async (client: AnyClient, defaultValues: any, data: any) => {
+  const referencedTypes = client[$modelRelationships];
+
+  if (!referencedTypes) {
+    throw new Error("No referenced types found");
+  }
+
+  function getUpdates(data: any): Record<string, number[]> {
```

Comment:
nit: do these functions need to be defined inline in the parent function? if not, can we move them out ot the module scope? redefining them each invocation consumes more memory and slows it down

Reply:
Yeah for sure, I inlined the functions since It felt like reshapeDataForGraphAPI is the only one that's gonna use it. I'm starting to wonder if maybe I should just create a "reshaper" class or something to encapsulate it all.